tests/components/pi_hole/test_config_flow.py

```diff
@@ -34,20 +31,19 @@ async def test_flow_user(hass: HomeAssistant):
         assert result["type"] == FlowResultType.FORM
         assert result["step_id"] == "user"
         assert result["errors"] == {}
-        _flow_next(hass, result["flow_id"])
 
         result = await hass.config_entries.flow.async_configure(
             result["flow_id"],
             user_input=CONF_CONFIG_FLOW_USER,
         )
         assert result["type"] == FlowResultType.FORM
-        assert result["step_id"] == "api_key"
-        assert result["errors"] is None
-        _flow_next(hass, result["flow_id"])
+        assert result["step_id"] == "user"
+        assert result["errors"] == {CONF_API_KEY: "invalid_auth"}
 
+        mocked_hole.data = ZERO_DATA
         result = await hass.config_entries.flow.async_configure(
             result["flow_id"],
-            user_input=CONF_CONFIG_FLOW_API_KEY,
+            user_input=CONF_CONFIG_FLOW_USER,
```

Comment:
Side note: We normally only name config key constants with a `CONF_` prefix. This is the whole config dict.

Reply:
yeah you are right ... will change it even it was "already there" before 👍